RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,907 @@
+  where each relation is an unordered collection of tuples.
+* **Required Attributes** A set, comprising the declared attributes that must exist for a data item to conform to the
+  Schema.
+* **Schema:** Schema is an abstract entity that is used as a blueprint to describe data along with its associated
+  constraints.
+* **Tuple:** The rows of relations which are collection of column values. In a Non-relational data model (E.g.
+  NoSQL/Semi-structured model) denotes a collection of name/value pairs also known as an Item. Tuples can be ordered or
+  unordered depending on the database system.
+
+*Note: Some examples and parts of the text are based on [PostgresSQL INSERT documentation.](https://www.postgresql.org/docs/current/sql-insert.html)*
+
+### Out of scope
+
+* Returning output from INSERT statement.
+* Constraints and unique index syntax semantics.
+* Type coercion rules.
+* Security privileges for execution of DML statements.
+
+## 2. Proposed Grammar and Semantics
+
+Figure 1. illustrate the BNF grammar corresponding to PartiQL’s `INSERT INTO ... ON CONFLICT` statements. 
+See Appendix 1. for the Grammar conventions.
+
+Note: all the definitions that are missing from the Grammar refer to the definitions that are already in SQL-92 standard.
+
+```EBNF
+<insert statement> ::=
+    INSERT INTO <table name> [ AS <alias> ] [  ( <attr name> [, <attr name> ]... ) ]
+        <value expr>
+    [ ON CONFLICT [ <conflict target> ] <conflict action> ]
+
+<value expr> ::= { DEFAULT VALUES | <values clause>
+      | <bag value> | <sub-select> }
+
+<values clause> ::= VALUES
+    ( { <expr> | DEFAULT } [, { <expr> | DEFAULT } ]...) [,...]
+
+<conflict target> ::=
+    ( { <index attr name> | <index expression> } [, ...] )
+      [ WHERE <index predicate> ]
+    | ( { <primary key> | <composite primary key> } )
+    | ON CONSTRAINT <constraint name>
+
+<conflict action> ::= DO NOTHING
+    | <do update>
+    | <do replace>
+
+<do update> ::= DO UPDATE
+    EXCLUDED
+    | SET {
+        {
+            <attr name> = { <expr> | DEFAULT }
+            | ( <attr name> [, <attr name>]... ) = ( { <expr> | DEFAULT } [, ...] )
+            | ( <attr name> [, <attr name>]... ) = ( <sub-select> )
+        } [, ...]
+        | <bag value>
+      }
+   [ WHERE <condition> ]
+
+<do replace> ::= DO REPLACE { <tuple value> | EXCLUDED }
+
+<primary key> ::= <attr name>
+
+<composite primary key> ::= <attr name> [, <attr name> ]
+
+(* The follwowing are derived directly from PartiQL's spec. *)
+<value> ::= <absent value>
+    | <scalar value>
+    | <tuple value>
+    | <collection value>
+
+<absent value ::= NULL
+    | MISSING
+
+<scalar value> :== <backtick> <ion literal> <backtick>
+    | sql literal
+
+<tuple value> ::= <l curly brace> <r curly brace>
+    | <l curly brace> <string value> : <value> [, <string value> : <value>]... <r curly brace>
```

Comment:
<string value> is not defined.

Reply:
Discussed this offline, will iterate over the grammar.